internal/discovery/foreign-cluster-operator/auth.go

```diff
@@ -69,18 +56,10 @@ func (r *ForeignClusterReconciler) getRemoteClient(
 	}
 
 	// not existing role
-	isPending := fc.Status.AuthStatus == discovery.AuthStatusPending
+	isPending := fc.Status.AuthStatus == discovery.AuthStatusPending || fc.Status.AuthStatus == ""
```

Comment:
Can we bypass this new condition by defaulting the status resource?

Reply:
I tried to set it as optional, both the field and the whole status struct in this way
```go
type ForeignClusterStatus struct {
	...
	// Authentication status.
	// +kubebuilder:validation:Enum="Pending";"Accepted";"Refused";"EmptyRefused"
	// +kubebuilder:default="Pending"
	// +kubebuilder:validation:Optional
	AuthStatus discovery.AuthStatus `json:"authStatus"`
}
```
```go
type ForeignCluster struct {
	metav1.TypeMeta   `json:",inline"`
	metav1.ObjectMeta `json:"metadata,omitempty"`

	Spec ForeignClusterSpec `json:"spec,omitempty"`
	// +kubebuilder:validation:Optional
	Status ForeignClusterStatus `json:"status"`
}
```

but it is still not working, since when a new ForeignCluster has been created it is in this form
```yaml
apiVersion: discovery.liqo.io/v1alpha1
kind: ForeignCluster
metadata:
  annotations:
    LastUpdate: "1623332772"
  creationTimestamp: "2021-06-10T13:45:23Z"
  generation: 1
  labels:
    discovery.liqo.io/cluster-id: a50a67a1-9986-47d1-9127-20e1dd5f5838
    discovery.liqo.io/discovery-type: LAN
  name: a50a67a1-9986-47d1-9127-20e1dd5f5838
  resourceVersion: "10133"
  selfLink: /apis/discovery.liqo.io/v1alpha1/foreignclusters/a50a67a1-9986-47d1-9127-20e1dd5f5838
  uid: 99c7ba81-1d3b-4b93-8011-6e9134efb941
spec:
  authUrl: https://172.18.0.3:31383
  clusterIdentity:
    clusterID: a50a67a1-9986-47d1-9127-20e1dd5f5838
  discoveryType: LAN
  join: true
  namespace: liqo
  trustMode: Untrusted
  ttl: 90
```

where the status is not present at all. With this resource, when the client retrieves the status from the API Server, it sees as nil and fills the authStatus as an empty string...

I propose to not change it at the moment since the field will completely be deleted in the refactoring proposed in #670 
@palexster @giorio94

Reply:
That's fine from my point of view. I'm honestly not sure it is possible to default status fields